### PR TITLE
7020 rebase - Propagate weights in FittingWithOutlierRemoval

### LINKED ISSUE #7019: FittingWithOutlierRemoval does not pass weights to underlying fitter

### PR DESCRIPTION
**Pull request by nden**
This is just a rebase of #7020. I rebased, squashed all commits and removed a duplicate change log entry. The original PR had merged from astropy/master and needed a rebase to pick up recent build fixes.

When merged this will close #7020 and fixes #7019.

The original one was already approved by @jehturner .

### CODE REVIEW

**Comment by nden** [comment]
This is too smart and won't let me approve a PR I created.

**Comment by jehturner** [comment]
Hmmm, I believe I don't have write access so my reviews are only advisory...

**Comment by nden** [comment]
@pllim Can you please approve this.

**Comment by pllim** [comment]
Before I do so, why is the bot still complaining about the change log?

File: CHANGES.rst [hunk 1]
```diff
@@ -430,6 +430,12 @@ astropy.modeling
 - Added ``is_separable`` function to modeling to test the
   separability of a model. [#6746]
 
+- Added ``Model.separable`` property. It returns a boolean value or
+  ``None`` if not set. [#6746]
+
+astropy.nddata
```
Comment (pllim) [inline CHANGES.rst]: This doesn't look right.

File: CHANGES.rst [hunk 2]
```diff
@@ -816,6 +822,7 @@ astropy.io.votable
 
 astropy.modeling
 ^^^^^^^^^^^^^^^^
+- Propagate weights to underlying fitter in ``FittingWithOutlierRemoval`` [#7020]
```
Comment (pllim) [inline CHANGES.rst]: Need to add `#7249` as well to the log.

File: CHANGES.rst (hunk 1 above)
Comment (nden) [reply CHANGES.rst]: Not sure what happened here. However this is probably OK but the entire entry for nddata (just below) looks wrong.

**Comment by bsipocz** [comment]
(@nden - in the future no need to open a new PR, you as a maintainer can force push the rebased version back to the original one)

**Comment by nden** [comment]
@bsipocz OK, thanks. I fear that if there were errors in the rebase (like in the new PR) I will push them and overwrite the original one. A new PR allows me to compare differences wiht the original PR. Is it a problem if a new PR is created after a rebase? For simple ones I certainly don't mind pusing over the original one.

**Comment by bsipocz** [comment]
@nden - No, there is no problem with opening a new PR, in fact I could equally argue for and against it. I just simply wasn't sure you knew about that option.

File: CHANGES.rst (hunk 2 above)
Comment (bsipocz) [reply CHANGES.rst]: please just have the number of this PR rather than as well, as the other PR is obsolete/duplicate once this one is merged.

**Comment by pllim** [comment]
I think coveralls is stuck. Merging this anyway since it had been previously approved.

**Comment by nden** [comment]
Thanks @pllim !

**Comment by jehturner** [comment]
The coverage must be the same or greater, with all those test cases and minimal new code...

**Pull request merged**